generator/internal/discovery_resource.cc

```diff
@@ -29,9 +29,11 @@ namespace generator_internal {
 
 DiscoveryResource::DiscoveryResource() : json_("") {}
 
-DiscoveryResource::DiscoveryResource(std::string name, std::string default_host,
+DiscoveryResource::DiscoveryResource(std::string name, std::string package_name,
```

Comment:
This many string parameters makes it really easy to swap them. Consider a struct or a small class.